src/axom/inlet/tests/inlet_object.cpp

```diff
@@ -77,6 +77,38 @@ TYPED_TEST(inlet_object, simple_struct_by_value)
   EXPECT_FALSE(foo.baz);
 }
 
+TYPED_TEST(inlet_object, simple_struct_verify_pass)
+{
+  std::string testString = "";
+  DataStore ds;
+  Inlet inlet = createBasicInlet<TypeParam>(&ds, testString);
+
+  auto& foo_table = inlet.addStruct("foo");
+  foo_table.addBool("bar", "bar's description").required(true);
+  foo_table.addBool("baz", "baz's description").required(true);
+
+  // Should pass verification as the struct is not present
+  EXPECT_TRUE(inlet.verify());
+}
+
+TYPED_TEST(inlet_object, simple_struct_verify_fail)
+{
+  std::string testString = "foo = { baz = true }";
+  DataStore ds;
+  Inlet inlet = createBasicInlet<TypeParam>(&ds, testString);
+
+  auto& foo_table = inlet.addStruct("foo");
+  foo_table.addBool("bar", "bar's description").required(true);
+  foo_table.addBool("baz", "baz's description").required(true);
+
+  // It should fail because a) the struct exists, and
+  // b) one of the required fields is not present
+  // Note that the only reason the struct is marked as present
+  // is because at least one of its expected fields is present, i.e.,
+  // having a foo = { quux = true } will pass verification
```

Comment:
This sounds wrong. Why would that pass verification?

Reply:
It fails because while the "foo" structure is optional, the fields "bar" and "baz" are marked required.  So the logic is if "foo" is present, it MUST have both "bar" and "baz". 

If there was no "foo" at all, it would pass verification.

Did i explain that better, @estebanpauli ?

If this is not already an example in the documentation it should be added.

Reply:
Ah you were referring to the comment.

Reply:
Can this be fixed in a follow-up PR?

```
foo = {}
```

Should also fail...

Reply:
Yes, fixing this requires #479 so it will have to be in a follow-up